lib/ansible/module_utils/vmware.py

```diff
@@ -354,13 +354,11 @@ def gather_vm_facts(content, vm):
         for device in vmnet:
             net_dict[device.macAddress] = list(device.ipAddress)
 
-    for dummy, v in iteritems(net_dict):
-        for ipaddress in v:
-            if ipaddress:
-                if '::' in ipaddress:
-                    facts['ipv6'] = ipaddress
-                else:
-                    facts['ipv4'] = ipaddress
+    if vm.guest.ipAddress:
+        if '::' in vm.guest.ipAddress:
```

Comment:
It looks like the original code could return both `ipv4` and `ipv6` addresses.

Where are with your change, only the first found of ipv6 or ipv4 would be returned.

Is it possible for this interface to have *both* IPv4 and IPv6 addresses?

Reply:
Using this property you are basically letting VMWare pick one address to throw at you so it could be either v4 or v6 but it will be only one address.

My change doesn't rely on checking interfaces it just asks VMWare for the answer directly.